### DISCUSSION TITLE
Order and order line screen

### DISCUSSION THREAD

**Discussion by Programmer-AC** [Budibase/budibase#7219] [Help]
I "discover" Budibase two days ago and playing around. Everything looks cool and easy to use. Thanks for your afford.
I find it is almost impossible to build traditional order and order line application.
Basic structure
1. Order ------(n)->OrderLine
2. OrderLine ------(1)-> Product

I have following problems
1. I cannot make 1 OrderLine point to 1 Product relationship
2. When I build the UI, I can create Order screen, I can create OrderLine screen but I cannot create a screen that top compartment is Order information and have OrderLine (of this order) in the bottom of the screen
3. I cannot create order line within the screen

I tested almost all the build-in template and cannot find any example close to above usage.

BTW, I tested the ticket system template and it cannot add follow-up to the ticket. This is not usable for daily usage.

**Comment by R2bEEaton** [comment]
Hi there @Programmer-AC! I am also a relatively new user of Budibase and found myself that they didn't have a template built for my exact use case. That's okay though! Budibase is very versatile and is actually built exactly for the kind of app you are envisioning.

Your basic structure can be achieved using relationships (as you mentioned). [Relationships](https://docs.budibase.com/docs/relationships) can be set in a number of different ways, including [relationship pickers](https://docs.budibase.com/docs/pickers#relationship-picker) or by setting the relationship column of one row to the `_id` of the desired related row. You can limit this to a one-to-many relationship using [Automation](https://docs.budibase.com/docs/automation-steps) logic for example.

Once you have your relationships set up properly, you can use a Data Provider to get the Order based on an OrderLine, or vice versa, or all the OrderLines based on one Order. This way you can get them to show up on the same screen, or show order information.

Budibase has a bit of a learning curve but once you've got it, I assure you you'll be creating an app in no time! Please let me know if you have any more specific questions for how to achieve the things mentioned above, because I assure you it is all possible!

**Reply by melohagan** [reply]
Thanks @R2bEEaton 
This may help as well: https://docs.budibase.com/docs/fetching-data#displaying-relational-data
